src/video/types.rs

```diff
@@ -107,15 +109,64 @@ impl VideoSourceType {
 }
 
 impl VideoEncodeType {
-    //TODO: use trait fromstr, check others places
     pub fn from_str(fourcc: &str) -> VideoEncodeType {
         let fourcc = fourcc.to_uppercase();
-        match fourcc.as_str() {
-            "H264" => VideoEncodeType::H264,
-            "MJPG" => VideoEncodeType::Mjpg,
-            "YUYV" => VideoEncodeType::Yuyv,
-            _ => VideoEncodeType::Unknown(fourcc),
+
+        let fourcc = match fourcc.as_str() {
+            "H264" => return VideoEncodeType::H264,
+            "MJPG" => return VideoEncodeType::Mjpg,
+            // TODO: We can implement a [GstDeviceMonitor](https://gstreamer.freedesktop.org/documentation/gstreamer/gstdevicemonitor.html?gi-language=c) and then this manual mapping between v4l's and gst's fourcc will not be neccessary anymore.
+            // A list of possible v4l fourcc from the Linux docs:
```

Comment:
I believe that my previous comment shows how to do it with gstreamer VideoFormatInfo

Reply:
And, also:
https://docs.rs/gstreamer-video/latest/gstreamer_video/enum.VideoFormat.html#method.to_fourcc